app/views/users/_user.html.erb

```diff
@@ -1,16 +1,11 @@
-<%unless current_user== user%>
+  <% unless current_user == user%>
   <li>
     Name: <%= user.name %>
     <span class="profile-link">
       <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
     </span>
 
-    <%unless current_user == user%>
-        <%unless Invitation.invitation_exists?(current_user.id, user.id) || Invitation.invitation_exists?(user.id, current_user.id)%>
-          <%= form_for(user, url: user_invitations_path(user), method: 'POST') do |f|%>
-          <%= f.submit "Follow"%>  
-          <%end%>
-        <%end%>
-    <%end%>
+    <%= check_user?(current_user, user)%>
+        <%= check_friendship(current_user, user) %>
   </li>
 <%end%>
```

Comment:
Kindly remove logic in your views. There should be no logic in your views. Kindly use helpers to remove all the logic in your views.
[Where do I put my code ](http://codefol.io/posts/Where-Do-I-Put-My-Code/)